core/src/main/java/io/snyk/plugins/artifactory/scanner/ScannerModule.java

```diff
@@ -57,16 +57,29 @@ public void scanArtifact(@Nonnull RepoPath repoPath) {
     validateLicenseIssues(testResult, repoPath);
   }
 
-  protected Optional<PackageScanner> getScannerForPackageType(String path) {
+  protected PackageScanner getScannerForPackageType(String path) {
     if (path.endsWith(".jar")) {
-      return Optional.of(mavenScanner);
-    } else if (path.endsWith(".tgz")) {
-      return Optional.of(npmScanner);
-    } else if (path.endsWith(".whl") || path.endsWith(".tar.gz") || path.endsWith(".zip") || path.endsWith(".egg")) {
-      return Optional.of(pythonScanner);
-    } else {
-      return Optional.empty();
+      if (configurationModule.getPropertyOrDefault(SCANNER_PACKAGE_TYPE_MAVEN).equals("true")) {
+        return mavenScanner;
+      }
+      throw new CannotScanException("Maven repository scanning is disabled.");
```

Comment:
I'd add a reason why
```suggestion
      throw new CannotScanException("Maven repository scanning is disabled by Snyk Plugin configuration.");
```

Reply:
Done.